pkg/virt-api/webhooks/validating-webhook/admitters/instancetype-admitter_test.go

```diff
@@ -47,6 +49,43 @@ var _ = Describe("Validating Instancetype Admitter", func() {
 		Expect(response.Allowed).To(BeFalse(), "Expected instancetype to not be allowed")
 		Expect(response.Result.Code).To(Equal(int32(http.StatusBadRequest)), "Expected error 400: BadRequest")
 	})
+
+	It("should reject specs with memory overcommit and hugepages", func() {
+		instancetypeObj.Spec = instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+			CPU: instancetypev1alpha2.CPUInstancetype{
+				Guest: uint32(1),
+			},
+			Memory: instancetypev1alpha2.MemoryInstancetype{
+				Guest:             resource.MustParse("128M"),
+				OvercommitPercent: -15,
+			},
+		}
+		ar := createInstancetypeAdmissionReview(instancetypeObj)
+		response := admitter.Admit(ar)
+
+		Expect(response.Allowed).To(BeFalse(), "Expected instancetype to not be allowed")
+	})
+
+	It("should reject negative memory overcommit values", func() {
+		instancetypeObj.Spec = instancetypev1alpha2.VirtualMachineInstancetypeSpec{
+			CPU: instancetypev1alpha2.CPUInstancetype{
+				Guest: uint32(1),
+			},
+			Memory: instancetypev1alpha2.MemoryInstancetype{
+				Guest:             resource.MustParse("128M"),
+				OvercommitPercent: 15,
+				Hugepages: &v1.Hugepages{
+					PageSize: "1Gi",
+				},
+			},
+		}
+		ar := createInstancetypeAdmissionReview(instancetypeObj)
+		ar.Request.Resource.Version = "unsupportedversion"
+		response := admitter.Admit(ar)
+
+		Expect(response.Allowed).To(BeFalse(), "Expected instancetype to not be allowed")
+		Expect(response.Result.Code).To(Equal(int32(http.StatusBadRequest)), "Expected error 400: BadRequest")
```

Comment:
This should assert on the error msg `overCommitPercent and hugepages should not be requested together.`.